java.lang.Exception: error when processing com.intellij.execution.process.ProcessEvent[source=bash -ic "latexmk -pdf -file-line-error -interaction=nonstopmode -synctex=1 -output-format=pdf -output-directory=/home/pi/repos/lab_elettronica/lab3/out -lualatex

### Type of JetBrains IDE (IntelliJ, PyCharm, etc.) and version
CLion 2025.1 (build CL-251.23774.442)

### Operating System 
Windows 11.0 (amd64)

### TeXiFy IDEA version
0.10.2

### Description
I'm not sure how this happened.
I was recompiling a file and (maybe) this happened when I closed the compilation tab of another file (well, the same file but it was from a previous run).
Anyway, the crash was "smooth", if it wasn't for the error popup I wouldn't have noticed anything.

### Stacktrace
```
java.lang.Exception: error when processing com.intellij.execution.process.ProcessEvent[source=bash -ic "latexmk -pdf -file-line-error -interaction=nonstopmode -synctex=1 -output-format=pdf -output-directory=/home/pi/repos/lab_elettronica/lab3/out -lualatex /home/pi/repos/lab_elettronica/lab3/main.tex"] by class nl.hannahsten.texifyidea.editor.autocompile.AutoCompileDoneListener
	at com.intellij.execution.process.ProcessHandler$2.processTerminated(ProcessHandler.java:268)
	at com.intellij.execution.process.ProcessHandler.lambda$notifyTerminated$2(ProcessHandler.java:207)
	at com.intellij.execution.process.ProcessHandler$TasksRunner.execute(ProcessHandler.java:325)
	at com.intellij.execution.process.ProcessHandler.notifyTerminated(ProcessHandler.java:190)
	at com.intellij.execution.process.ProcessHandler.notifyProcessTerminated(ProcessHandler.java:186)
	at com.intellij.execution.process.BaseProcessHandler.onOSProcessTerminated(BaseProcessHandler.java:74)
	at com.intellij.execution.process.OSProcessHandler.onOSProcessTerminated(OSProcessHandler.java:188)
	at com.intellij.execution.process.BaseOSProcessHandler$1.lambda$startNotified$0(BaseOSProcessHandler.java:82)
	at com.intellij.execution.process.ProcessWaitFor.lambda$new$0(ProcessWaitFor.java:45)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:249)
	at com.intellij.execution.process.ProcessWaitFor.lambda$new$1(ProcessWaitFor.java:23)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:102)
	at com.intellij.util.concurrency.ChildContext$runInChildContext$1.invoke(propagation.kt:102)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:108)
	at com.intellij.util.concurrency.ChildContext.runInChildContext(propagation.kt:102)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:735)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:732)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:400)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:732)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.intellij.openapi.diagnostic.RuntimeExceptionWithAttachments: Access is allowed from Event Dispatch Thread (EDT) only; If you access or modify model on EDT consider wrapping your code in WriteIntentReadAction  or ReadAction; see https://jb.gg/ij-platform-threading for details
Current thread: Thread[#1371,bash,4,main] 1890035011 (EventQueue.isDispatchThread()=false)
SystemEventQueueThread: Thread[#40,AWT-EventQueue-0,10,main] 743091915
	at com.intellij.util.concurrency.ThreadingAssertions.createThreadAccessException(ThreadingAssertions.java:257)
	at com.intellij.util.concurrency.ThreadingAssertions.throwThreadAccessException(ThreadingAssertions.java:248)
	at com.intellij.util.concurrency.ThreadingAssertions.assertEventDispatchThread(ThreadingAssertions.java:89)
	at com.intellij.openapi.ui.DialogWrapper.<init>(DialogWrapper.java:248)
	at com.intellij.openapi.ui.DialogWrapper.<init>(DialogWrapper.java:239)
	at com.intellij.ui.messages.AlertDialog.<init>(AlertMessagesManager.kt:108)
	at com.intellij.ui.messages.AlertMessagesManager.showMessageDialog(AlertMessagesManager.kt:70)
	at com.intellij.ui.messages.MessagesServiceImpl.showMessageDialog(MessagesServiceImpl.java:56)
	at com.jetbrains.rdserver.ui.BackendMessagesService.showMessageDialog(BackendMessagesService.kt:38)
	at com.intellij.openapi.ui.messages.MessagesService.showMessageDialog$default(MessagesService.kt:53)
	at com.intellij.openapi.ui.OkCancelDialogBuilder.show(MessageDialogBuilder.kt:215)
	at com.intellij.openapi.ui.OkCancelDialogBuilder.ask(MessageDialogBuilder.kt:205)
	at com.intellij.openapi.ui.Messages.showOkCancelDialog(Messages.java:485)
	at com.intellij.execution.impl.ExecutionManagerImplKt.userApprovesStopForSameTypeConfigurations(ExecutionManagerImpl.kt:1016)
	at com.intellij.execution.impl.ExecutionManagerImplKt.access$userApprovesStopForSameTypeConfigurations(ExecutionManagerImpl.kt:1)
	at com.intellij.execution.impl.ExecutionManagerImpl.restartRunProfile(ExecutionManagerImpl.kt:570)
	at com.intellij.execution.impl.ExecutionManagerImpl.restartRunProfile(ExecutionManagerImpl.kt:537)
	at com.intellij.execution.ExecutionManager.restartRunProfile(ExecutionManager.kt:100)
	at nl.hannahsten.texifyidea.editor.autocompile.AutoCompileState.scheduleCompilation(AutoCompileState.kt:80)
	at nl.hannahsten.texifyidea.editor.autocompile.AutoCompileState.scheduleCompilationIfNecessary(AutoCompileState.kt:51)
	at nl.hannahsten.texifyidea.editor.autocompile.AutoCompileDoneListener.processTerminated(AutoCompileDoneListener.kt:13)
	at com.intellij.execution.process.ProcessHandler$2.processTerminated(ProcessHandler.java:262)
	... 24 more

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,8 @@
 * Add filters to filter out some elements by default in the file structure popup
 * Add setting to enable spellcheck in 'code' scope
 
-### Fixed
+### Fixed  
+* Make sure not to start a new autocompile run if a previous one is still running
 * Fix validation when renaming a bibtex identifier
 * Fix a possible UI freeze when computing obsolete pdf viewer settings
 * Do not run grammar checks in graphicspath argument

--- a/src/nl/hannahsten/texifyidea/editor/autocompile/AutoCompileState.kt
+++ b/src/nl/hannahsten/texifyidea/editor/autocompile/AutoCompileState.kt
@@ -8,6 +8,7 @@ import com.intellij.notification.Notification
 import com.intellij.notification.NotificationType
 import com.intellij.openapi.project.Project
 import nl.hannahsten.texifyidea.run.latex.LatexRunConfiguration
+import java.util.concurrent.atomic.AtomicBoolean
 
 /**
  * State of autocompilation.
@@ -17,7 +18,7 @@ import nl.hannahsten.texifyidea.run.latex.LatexRunConfiguration
 object AutoCompileState {
 
     /** Whether an autocompile is in progress. */
-    private var isCompiling = false
+    private var isCompiling = AtomicBoolean(false)
 
     /** Whether the document has changed since the last triggered autocompile. */
     private var hasChanged = false
@@ -33,7 +34,7 @@ object AutoCompileState {
         this.project = project
 
         // Remember that the document changed, so a compilation should be scheduled later
-        if (isCompiling) {
+        if (isCompiling.get()) {
             hasChanged = true
         }
         else {
@@ -51,7 +52,7 @@ object AutoCompileState {
             scheduleCompilation()
         }
         else {
-            isCompiling = false
+            isCompiling.set(false)
         }
     }
 
@@ -61,9 +62,6 @@ object AutoCompileState {
 
             return
         }
-
-        isCompiling = true
-        hasChanged = false
 
         // Get run configuration selected in the combobox and run that one
         if (project!!.isDisposed) return
@@ -77,6 +75,11 @@ object AutoCompileState {
         // Changing focus would interrupt the user during typing
         (runConfigSettings.configuration as LatexRunConfiguration).allowFocusChange = false
 
+        hasChanged = false
+        // Ensure we only trigger one compilation at a time
+        if (isCompiling.getAndSet(true)) return
+
+        // If the run config is already running, this may trigger a dialog asking the user whether to stop and rerun
         ExecutionManager.getInstance(project!!).restartRunProfile(
             project!!,
             DefaultRunExecutor.getRunExecutorInstance(),

